packages/ragbits-document-search/src/ragbits/document_search/retrieval/rephrasers/llm.py

```diff
@@ -0,0 +1,64 @@
+from typing import Optional
+
+from ragbits.core.llms import get_llm
+from ragbits.core.llms.base import LLM, LLMOptions
+from ragbits.core.utils.config_handling import get_cls_from_config
+from ragbits.document_search.retrieval import rephrasers
+from ragbits.document_search.retrieval.rephrasers.base import QueryRephraser
+from ragbits.document_search.retrieval.rephrasers.prompt_query_rephraser import QueryRephraserPrompt, _PromptInput
+
+
+class LLMQueryRephraser(QueryRephraser):
+    """A rephraser class that uses a LLM to rephrase queries."""
+
+    def __init__(self, llm: LLM, prompt: Optional[QueryRephraserPrompt] = None):
+        """
+        Initialize the LLMQueryRephraser with a LLM and an optional prompt.
+
+        Args:
+            llm: A LLM instance to handle query rephrasing.
+            prompt: A prompt defining how the rephrasing should be done.
+                If not provided, the default `QueryRephraserPrompt` is used.
+        """
+
+        self._prompt = prompt or QueryRephraserPrompt
```

Comment:
Wouldn't this mean that `_prompt` sometimes is an _instance_ of `QueryRephraserPrompt` and sometimes it's the class itself? I also don't see `self._prompt` used anywhere. In the `rephrase` method  `QueryRephraserPrompt` is used directly.

Reply:
Right, I cleaned it up now and prompt is no longer an attribute